### ISSUE TITLE
NativeAOT compiler assert on static abstract GVM

### ISSUE THREAD

**Issue by vitek-karas** [dotnet/runtime#78116]
Sample code to repro:

```C#
	public class StaticGVMRepro
	{
		public static void Main()
		{
			typeof (Impl).RequiresPublicMethods ();
		}

		public interface IAnnotatedMethods
		{
			static abstract void GenericWithMethodStatic<T> ();  // This is what's causing the assert

			void InstanceGVM<T> (); // This needs to be here so that it triggers GVM processing for the type
		}

		public class Impl : IAnnotatedMethods
		{
			public static void GenericWithMethodStatic<T> () { }

			public void InstanceGVM<T> () { }
		}
	}
```

The assert happens here: https://github.com/dotnet/runtime/blob/af494f4060cdc03782d0b4cf68bf073a4d979260/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeGVMEntriesNode.cs#L146

**Comment by vitek-karas** [comment]
I "think" the fix is to turn the assert into an if and simply skip static virtual methods. But I actually don't know how we handle static GVMs in the compiler, so maybe it actually needs the code around the assert to happen.

@MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
For now I think just `continue` is enough. They don't participate in the analysis because they're resolved at compile time right now.

Fortunately, it looks like release builds are going to do the right thing (try to resolve, fail, and skip) so we won't need to backport.

**Comment by MichalStrehovsky** [comment]
Fixed in #78157.
